### PR TITLE
SCRUM-87-sign-up update: auth endpoint fixes

### PR DESCRIPTION
**Pull request by somT-oss**

### CODE REVIEW

File: agents/serializer.py [hunk 1]
```diff
@@ -125,9 +125,11 @@ class AgentForgotPasswordResetSerializer(serializers.Serializer):
...
-        new_password = self.validated_data.get('new_password')
-        confirm_password = self.validated_data.get('confirm_password')
+    def validate(self, attrs):
+        new_password = attrs.get('new_password')
+        confirm_password = attrs.get('confirm_password')
 
         if new_password != confirm_password:
-            return ValidationError("Password Mismatch")
+            raise ValidationError("Password Mismatch")
+
```
Comment (50-Course) [inline agents/serializer.py:134]: While there is no problem with this logic itself, I suggest you should keep an `errors.py` file or an `exception.py` for each business entity, `domain-specific` apps eg this app `agents` for example. so you can reference them as `agents.errors` or `agents.exceptions`.

File: agents/views.py [hunk 2]
```diff
@@ -38,17 +38,20 @@ def create_agent(request) -> Response:
         last_name = serializer_class.validated_data.get('last_name')
         agent_email = serializer_class.validated_data.get('email')
```
Comment (50-Course) [inline agents/views.py:39]: `validated_data.get()` here would default to `NoneType`, which would fault the use of `try-catch` block.


`validated_data['key']` is the correct approach, that would raise a `KeyError` for missing keys in the serializer, and that is what you are essentially capturing here in the `Exception` i believe?

File: agents/serializer.py (hunk 1 above)
Comment (somT-oss) [reply agents/serializer.py:134]: I'll look into this. Thank you for your observation

File: agents/views.py (hunk 2 above)
Comment (somT-oss) [reply agents/views.py:39]: Yeah. I noticed this during testing the endpoint.
I'll make the change.

File: insurer/serializer.py [hunk 3]
```diff
@@ -111,9 +111,11 @@ class ForgotPasswordResetSerializer(serializers.Serializer):
...
-    def check_passwords_equal(self) -> ValidationError:
-        new_password = self.validated_data.get('new_password')
-        confirm_password = self.validated_data.get('confirm_password')
+    def validate(self, attrs):
+        new_password = attrs.get('new_password')
+        confirm_password = attrs.get('confirm_password')
 
         if new_password != confirm_password:
-            return ValidationError("Password Mismatch")
+            raise ValidationError("Password Mismatch")
```
Comment (50-Course) [inline insurer/serializer.py:119]: Just a quick reminder, `Insurer` and `Agents` seem to have this same `validate_password` flow. You should keep this in a `utils.py` package, and have them called here under the `validate` method of the serializer. 

and give this `utils.py` package a custom exception class, e.g:

```python
class PasswordMismatchError(Exception):
...
```

then in your `utils.py` you raise that, and your `validate` function looks similar to this:

```python
class Agent...Serializer(serializers.Serializer):
   def validate(self, attrs: dict[str, object]) -> bool:
       return validate_password(attrs)
```

**Review by 50-Course — COMMENTED** [review]
Hi @somT-oss,

Great work here, but permit me to give a little room for improvement here. Please review my suggestions, thank you. 

You can ignore the types for the moment, soon the repo can adopt typing after the overall goal is met.

**Comment by somT-oss** [comment]
I definitely would.
Thanks for the suggestions. I'm picking up one or two things already

**Comment by 50-Course** [comment]
With that, you may request from a second pass, from @Mbaoma or @izu-stanley before merge. Probably, Omar because of Izu's schedules.

File: agents/views.py (hunk 2 above)
Comment (izu-stanley) [reply agents/views.py:39]: We could add a default type perhaps? 

DEFAULT_NAME=-1
validation.get('name', DEFAULT_NAME)

What do you think? I don't have full context of the change so open to suggestions

**Pull request merged**